overlord/snapstate/target.go

```diff
@@ -413,14 +414,12 @@ func completeStoreAction(action *store.SnapAction, revOpts RevisionOptions, igno
 
 		// if the caller didn't provide any validation sets, make sure that
 		// the snap is allowed by all of the enforced validation sets
-		invalidSets, err := vsets.CheckPresenceInvalid(naming.Snap(action.InstanceName))
+		pres, err := vsets.Presence(naming.Snap(action.InstanceName))
```

Comment:
Will this get the right presence if we use an instance name?

Reply:
Good catch, thank you.

Reply:
This is fixed in d139fc45db1e4221e27899affd008f17a8a3d66e and 4d28fdab06364c7e9820b400b3dda255fc494417.